ballerina/schema-client.bal

```diff
@@ -0,0 +1,319 @@
+    #             information will be retrieved
+    # + return - An 'TableDefinition' with the relevant table information or an `sql:Error`
+    isolated remote function getTableInfo(string tableName, sql:ColumnRetrievalOptions include = sql:COLUMNS_ONLY) returns TableDefinition|sql:Error {
+        record {}|sql:Error 'table = self.dbClient->queryRow(
+            `SELECT TABLE_TYPE FROM INFORMATION_SCHEMA.TABLES 
+             WHERE (TABLE_SCHEMA=${self.database} AND TABLE_NAME = ${tableName});`
+        );
+
+        if 'table is sql:NoRowsError {
+            return error sql:NoRowsError("The selected table does not exist or the user does not have the required privilege level to view the table.");
+        } else if 'table is sql:Error {
+            return 'table;
+        } else {
+            TableDefinition tableDef = {
+                name: tableName,
+                'type: <sql:TableType>'table["TABLE_TYPE"]
+            };
+
+            if !(include == sql:NO_COLUMNS) {
+                TableDefinition|sql:Error tableDefError = self.getColumns(tableName, tableDef);
+        
+                if tableDefError is TableDefinition {
+                    tableDef = tableDefError;
+                }
+
+                if include == sql:COLUMNS_WITH_CONSTRAINTS {
+                    tableDefError = self.getConstraints(tableName, tableDef);
+
+                    if tableDefError is TableDefinition {
+                        tableDef = tableDefError;
+                    }
+                }    
+            }
+
+            return tableDef;
+        }
+    }
+
+    # Retrieves all routines in the database.
+    #
+    # + return - A string array containing the names of the routines or an `sql:Error`
+    isolated remote function listRoutines() returns string[]|sql:Error {
+        string[] routines = [];
+        stream<record {}, sql:Error?> results = self.dbClient->query(
+            `SELECT ROUTINE_NAME FROM INFORMATION_SCHEMA.ROUTINES
+            WHERE ROUTINE_SCHEMA = ${self.database};`
+        );
+
+        do {
+            routines = check from record {} result in results
+                select <string>result["ROUTINE_NAME"];
+        } on fail error e {
+            return error(string `Error while listing routines in the ${self.database} database.`, cause = e);
+        }
+
+        check results.close();
+
+        return routines;
+    }
+
+    # Retrieves information relevant to the provided routine in the database.
+    #
+    # + name - The name of the routine
+    # + return - An 'sql:RoutineDefinition' with the relevant routine information or an `sql:Error`
+    isolated remote function getRoutineInfo(string name) returns sql:RoutineDefinition|sql:Error {
+        sql:RoutineDefinition|sql:Error routine = self.dbClient->queryRow(
+            `SELECT ROUTINE_TYPE, DATA_TYPE FROM INFORMATION_SCHEMA.ROUTINES 
+             WHERE ROUTINE_NAME = ${name};`
+        );
```

Comment:
Does this get mapped to [`sql:RoutineDefinition`](https://github.com/ballerina-platform/module-ballerina-sql/blob/master/ballerina/metadata_types.bal#L119) correctly?

It seems the field names of the record and the column names are different

Reply:
The following lines 302 - 307 ensure that the mapping is correct
`        sql:RoutineDefinition routineDef = {
            name: name,
            'type: <sql:RoutineType>routine["ROUTINE_TYPE"],
            returnType: <string>routine["DATA_TYPE"],
            parameters: parameterList
        };
`

Reply:
Update: moved the mapping from getParamters to getRoutineInfo